### ISSUE TITLE
Why does pip install deepspeed fail to install?

### ISSUE THREAD

**Issue by PHPDUAN** [microsoft/DeepSpeed#5312]
I installed cuda11.8 and set the environment variables：
`(qwenpro) ya@C4090-2:~/qwenpro$ nvcc -V
nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2022 NVIDIA Corporation
Built on Wed_Sep_21_10:33:58_PDT_2022
Cuda compilation tools, release 11.8, V11.8.89
Build cuda_11.8.r11.8/compiler.31833905_0
(qwenpro) ya@C4090-2:~/qwenpro$ dpkg -l | grep cudnn
ii  cudnn-local-repo-ubuntu2204-8.9.7.29              1.0-1                                   amd64        cudnn-local repository configuration files
(qwenpro) ya@C4090-2:~/qwenpro$ pip show torch
Name: torch
Version: 2.2.1
Summary: Tensors and Dynamic neural networks in Python with strong GPU acceleration
Home-page: https://pytorch.org/
Author: PyTorch Team
Author-email: packages@pytorch.org
License: BSD-3
Location: /home/ya/anaconda3/envs/qwenpro/lib/python3.10/site-packages
Requires: filelock, fsspec, jinja2, networkx, nvidia-cublas-cu12, nvidia-cuda-cupti-cu12, nvidia-cuda-nvrtc-cu12, nvidia-cuda-runtime-cu12, nvidia-cudnn-cu12, nvidia-cufft-cu12, nvidia-curand-cu12, nvidia-cusolver-cu12, nvidia-cusparse-cu12, nvidia-nccl-cu12, nvidia-nvtx-cu12, sympy, triton, typing-extensions
Required-by: accelerate, optimum, peft, trl
`

Implementation feedback:
`(qwenpro) ya@C4090-2:~/qwenpro$ pip install deepspeed
Collecting deepspeed
  Using cached deepspeed-0.14.0.tar.gz (1.3 MB)
  Preparing metadata (setup.py) ... error
  error: subprocess-exited-with-error
  
  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [16 lines of output]
      Traceback (most recent call last):
        File "<string>", line 2, in <module>
        File "<pip-setuptools-caller>", line 34, in <module>
        File "/tmp/pip-install-iabcb2xp/deepspeed_807397c72531479a8160e4ef12374a51/setup.py", line 100, in <module>
          cuda_major_ver, cuda_minor_ver = installed_cuda_version()
        File "/tmp/pip-install-iabcb2xp/deepspeed_807397c72531479a8160e4ef12374a51/op_builder/builder.py", line 52, in installed_cuda_version
          output = subprocess.check_output([cuda_home + "/bin/nvcc", "-V"], universal_newlines=True)
        File "/home/ya/anaconda3/envs/qwenpro/lib/python3.10/subprocess.py", line 421, in check_output
          return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
        File "/home/ya/anaconda3/envs/qwenpro/lib/python3.10/subprocess.py", line 503, in run
          with Popen(*popenargs, **kwargs) as process:
        File "/home/ya/anaconda3/envs/qwenpro/lib/python3.10/subprocess.py", line 971, in __init__
          self._execute_child(args, executable, preexec_fn, close_fds,
        File "/home/ya/anaconda3/envs/qwenpro/lib/python3.10/subprocess.py", line 1863, in _execute_child
          raise child_exception_type(errno_num, err_msg, err_filename)
      FileNotFoundError: [Errno 2] No such file or directory: ':/usr/local/cuda-11.8:/usr/local/cuda-11.8:/usr/local/cuda-11.8:/usr/local/cuda-11.8:/usr/local/cuda-11.8:/usr/local/cuda-11.8:/usr/local/cuda-11.8/bin/nvcc'
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.
`
![截图 2024-03-26 14-40-06](https://github.com/microsoft/DeepSpeed/assets/36032958/c7a6bd48-71a5-4073-8cb2-a30bb1259d38)
![截图 2024-03-26 14-40-13](https://github.com/microsoft/DeepSpeed/assets/36032958/6e0e78cc-b838-4aac-9c51-7d98323b528b)

**Comment by loadams** [comment]
Hi @PHPDUAN - just checking, you've installed a version of torch that has cuda support?  So the install would look something like `pip install torch==2.1.2+cu118`?  But the underlying issue looks like DeepSpeed cannot find your cuda install?  Do you have a `CUDA_HOME` path set in your environment variables?

**Comment by AlongWY** [comment]
This is a reason that why i want  make pre-built wheels. It's too difficult to configure the compile environment.

**Comment by loadams** [comment]
@AlongWY - the JIT compilation environment and the pre-building ops environment should be the same though, so in order to run DeepSpeed on a machine you'll need to do that setup either way, right?

**Comment by AlongWY** [comment]
Yes, but the pre-built wheels don't need JIT, it needs only CUDA driver

**Comment by loadams** [comment]
Hi @PHPDUAN - can you share the output from the above commands?

**Comment by loadams** [comment]
Hi @PHPDUAN - closing this issue as it is stale now.  If you are still hitting this, please comment and we can re-open or open a new issue with the details requested above, thanks
